pandas/tests/indexing/test_indexing.py

```diff
@@ -153,7 +153,6 @@ def test_setitem_ndarray_3d(self, index, obj, idxr, idxr_id):
                     "floating",
                     "string",
                     "datetime64",
-                    "period",
```

Comment:
what changed here?

Reply:
on the `Series.__setitem__` case it now (correctly, i think) raises, as it is trying to instantiate an index with 3D data

Reply:
this test is a mess, is hiding the fact that all of the cases in this block _should_ raise but currently dont  (xref #31813)